pkg/upgrade/upgrades/backfill_jobs_info_table_migration_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package upgrades_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/keyvisualizer"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/server"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
+	"github.com/cockroachdb/cockroach/pkg/upgrade/upgradebase"
+	"github.com/cockroachdb/cockroach/pkg/upgrade/upgrades"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/stretchr/testify/require"
+)
+
+// TestBackfillJobsInfoTable tests that the `system.job_info` table is
+// backfilled with the progress and payload of each job in the `system.jobs`
+// table.
+func TestBackfillJobsInfoTable(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	clusterArgs := base.TestClusterArgs{
+		// Disable all automatic jobs creation and adoption.
+		ServerArgs: base.TestServerArgs{
+			DisableSpanConfigs: true,
+			Knobs: base.TestingKnobs{
+				// Avoiding jobs to be adopted.
+				JobsTestingKnobs: &jobs.TestingKnobs{
+					DisableAdoptions: true,
+				},
+				// DisableAdoptions needs this.
+				UpgradeManager: &upgradebase.TestingKnobs{
+					DontUseJobs:                       true,
+					SkipJobMetricsPollingJobBootstrap: true,
+				},
+				KeyVisualizer: &keyvisualizer.TestingKnobs{
+					SkipJobBootstrap: true,
+				},
+				Server: &server.TestingKnobs{
+					DisableAutomaticVersionUpgrade: make(chan struct{}),
+					BinaryVersionOverride: clusterversion.ByKey(
+						clusterversion.V23_1CreateSystemJobInfoTable - 1),
```

Comment:
nit: I'd just use V22_2 instead of the specific intermediate version - 1. the -1'ing like this is annoying if/when someone tries to delete the migration immediately prior to this one, since it'll panic with out-of-bounds at runtime. Basically, even though you'r'e not explicitly mentioning that prior version by name, you are obliquely referring to it in that you are assuming there _is_ a prior version at all; just naming a prior version to upgrade from explicitly makes this obvious (to grep/goland/at compile time), and 22_2 is as good a prior version as any to name.